### ISSUE TITLE
Feature Request: Support for MultiIndex DataFrames

### ISSUE THREAD

**Issue by kschutter** [microsoft/vscode-data-wrangler#311]
Currently, the extension concatenates multiple row indices into one index like 'layer1,layer2,layer3'. It would be amazing if the data was displayed similar to how vscode jupyter inline displays them.
DataWrangler: ![Image](https://github.com/user-attachments/assets/9bd312d6-5c94-4475-841d-8159e179de22)
Vscode Jupyter Inline: ![Image](https://github.com/user-attachments/assets/34a72150-122a-4823-b3df-1e3424492b3f)

**Comment by pwang347** [comment]
Hi @kschutter, thanks for opening this feature request! We have noted it and added it to our backlog.

**Comment by CharlesCheckley** [comment]
I'd also find this really useful

**Comment by notuntoward** [comment]
Yes, I use multi-index dataframes all the time, but never used datawrangler to look at them.  Then, after the latest datawrangler update, I was horrified when it took over all cell output dataframe display.  This made whole notebooks unreadable.  

But I was relieved that I could get rid of it by unclicking experimental datawrangler settings (screenshot below, in case anybody is hunting for this.).

It's unfortunate that I had to totally shut off cell output display.   I can see how useful it would be for non-multi-indexed dataframes.

![Image](https://github.com/user-attachments/assets/f66d59b2-1124-4876-ba18-aacdfd0ed531)

**Comment by etienneschalk** [comment]
Hello @pwang347 ,

I comment here my response to https://github.com/microsoft/vscode-data-wrangler/issues/488#issuecomment-2837157886

I confirm that the issue I originally created ( #488 ) is indeed similar to this issue.

> I assume we can generally align the behaviour to what Pandas does?

Yes exactly, I see Data Wrangler as an enhancer of the tradition pandas HTML display. It should add functionality on top of it, and multi index is part of it. 

For now, I call `.reset_index()` on my dataframe before displaying them when using Data Wrangler.

> Out of curiosity, would you say this is roughly the desired behaviour for the prefix merging?

>   Always show prefix for the first entry on a page

Yes, when we browse to the next page, having the whole multi index as a reminder would be better. Each page should behave like the existing HTML display. 

>    If out of order (e.g., due to sorting on a different column), repeat the prefix

I am not sure I get this. Is it related to a Data Wrangler feature, like sorting columns? I did not use it that much so I cannot really give feedback. But, information on a page should always be exhaustive, so I would say if for some reason the multi index becomes shuffled, information completeness should be the priority even if it leads to some repetition.

**Comment by MaxJ58** [comment]
I second @PlayfulPiano 's request above - having multi-indices displayed as separate columns that we can apply filters to would greatly improve the functionality. The best current workaround is to .reset_index() as @etienneschalk suggests.

**Comment by b-a0** [comment]
Note that columns could also have MultiIndexes, I would find it very helpful if that would be supported as well.
